data/migrations/1_create_msmt_results.sql

```diff
@@ -11,36 +13,155 @@ DROP TABLE `measurements`;
+
+    -- This is the measurement failed to run and the user should be offerred to
+    -- re-run it.
+    `is_failed` TINYINT(1) NOT NULL,
+    `failure_msg` VARCHAR(255),
+
+    `is_upload_failed` TINYINT(1) NOT NULL,
+    `upload_failure_msg` VARCHAR(255),
+
+    -- Is used to indicate that this particular measurement has been re-run and
+    -- therefore the UI can take this into account to either hide it from the
+    -- result view or at the very least disable the ability to re-run it.
+    -- XXX do we also want to have a reference to the re-run measurement?
+    `is_rerun` TINYINT(1) NOT NULL,
+
+    -- This is the server-side report_id returned by the collector. By using
+    -- report_id & input, you can query the api to fetch this measurement.
+    -- Ex.
+    -- GET https://api.ooni.io/api/v1/measurements?input=$INPUT&report_id=$REPORT_ID
+    -- Extract the first item from the `result[]` list and then fetch:
+    -- `measurement_url` to get the JSON of this measurement row.
+    -- These two values (`report_id`, `input`) are useful to fetch a
+    -- measurement that has already been processed by the pipeline, to
+    -- implement cleanup of already uploaded measurements.
+    `report_id` VARCHAR(255), -- This can be NULL when no report file has been
+                              -- created.
+
+    `url_id` INTEGER NOT NULL,
+    FOREIGN KEY (`url_id`) REFERENCES `urls`(`id`),
+
+    -- This is not yet a feature of the collector, but we are planning to add
+    -- this at some point in the near future.
+    -- See: https://github.com/ooni/pipeline/blob/master/docs/ooni-uuid.md &
+    -- https://github.com/ooni/pipeline/issues/48
+    `measurement_id` INT(64),
+
+    -- This indicates in the case of a websites test, that a site is likely
+    -- blocked, or for an IM test if the IM tests says the app is likely
+    -- blocked, or if a middlebox was detected.
+    -- You can `JOIN` a `COUNT()` of this value in the results view to get a count of
+    -- blocked sites or blocked IM apps
+    `is_anomaly` TINYINT(1),
+
+    -- This is an opaque JSON structure, where we store some of the test_keys
+    -- we need for the measurement details views and some result views (ex. the
+    -- upload/download speed of NDT, the reason for blocking of a site,
+    -- etc.)
+    `test_keys` JSON,
+
+    -- The cross table reference to JOIN the two tables together.
+    `result_id` INTEGER NOT NULL,
+    FOREIGN KEY (`result_id`) REFERENCES `results`(`id`)
+      ON DELETE CASCADE ON UPDATE CASCADE, -- If we delete a result we also want
+                                           -- all the measurements to be deleted as well.
+
+    -- This is a variable used internally to track the path to the on-disk
+    -- measurements.json. It may make sense to write one file per entry by
+    -- hooking MK and preventing it from writing to a file on disk which may
+    -- have many measurements per file.
+    `report_file_path` VARCHAR(260) NOT NULL,
```

Comment:
Also this can be removed as in the app we use the format `performance-Measurement_id.json`

Reply:
Same as for the above.